pkg/systems/smr/appmodule.go

```diff
@@ -30,8 +31,9 @@ type AppModule struct {
 }
 
 // NewAppModule creates a new AppModule.
-func NewAppModule(appLogic AppLogic, transport net.Transport, protocolModule t.ModuleID) *AppModule {
+func NewAppModule(ctx context.Context, appLogic AppLogic, transport net.Transport, protocolModule t.ModuleID) *AppModule {
```

Comment:
I'd prefer not using the context here and keep the TODO context when connecting the transport. That TODO context should eventually disappear and we don't have to pollute the `NewAppModule` interface.
Plus, it tends to be bad practice to store context in state variables.

Reply:
TODO context should disappear or the context should disappear at all?

Reply:
Can we address this context properly now but not eventually?

Reply:
Fixed. Removed all contexts